src/FakeItEasy/Configuration/AfterMustHaveHappenedConfiguration.cs

```diff
@@ -17,11 +17,38 @@ public AfterMustHaveHappenedConfiguration(FakeManager fakeManager, ICallMatcher
             this.repeatConstraint = repeatConstraint;
         }
 
-        public void InOrder(ISequentialCallContext context)
+        /// <summary>
+        /// Checks that the asserted call happened in order relative to others in the assertion chain.
+        /// </summary>
+        /// <param name="nextAssertion">An assertion describing the next call that should occur.</param>
+        /// <returns>An object that can be used to assert that a following call was made in the expected order.</returns>
+        /// <exception cref="ExpectationException">The call was not made in the expected order.</exception>
+        public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextAssertion)
         {
-            Guard.AgainstNull(context, "context");
+            var context = ServiceLocator.Current.Resolve<SequentialCallContext>();
+            this.CheckCallHappenedInOrder(context);
+            return new SubsequentOrderedCallAssertion(context).Then(nextAssertion);
+        }
 
+        private void CheckCallHappenedInOrder(SequentialCallContext context)
+        {
             context.CheckNextCall(this.fakeManager, this.matcher.Matches, this.callDescription, this.repeatConstraint);
         }
+
+        private class SubsequentOrderedCallAssertion : IOrderableCallAssertion
+        {
+            private readonly SequentialCallContext context;
+
+            public SubsequentOrderedCallAssertion(SequentialCallContext context)
+            {
+                this.context = context;
+            }
+
+            public IOrderableCallAssertion Then(IAfterMustHaveHappenedConfiguration nextCall)
+            {
+                ((AfterMustHaveHappenedConfiguration)nextCall).CheckCallHappenedInOrder(this.context);

```

Comment:
@adamralph, the parameter value differs because I'd had all `nextCall`s and then saw your multi-step example where you called these items "assertions" and I missed one class in the conversion. Well spotted. Will make them all assertions.